xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp

```diff
@@ -233,7 +233,7 @@ bool CActiveAEBufferPoolResample::Create(unsigned int totaltime, bool remap, boo
       if (m_processor->GetChannelLayout().Count() > 2)                    /* Disable upmix for CActiveAEResample if DSP layout > 2.0, becomes perfomed by DSP */
         upmix = false;
 
-      m_dspBuffer = new CActiveAEBufferPool(m_inputFormat);               /* Get dsp processing buffer class, based on dsp output format */
+      m_dspBuffer = new CActiveAEBufferPool(m_format);               /* Get dsp processing buffer class, based on dsp output format */
```

Comment:
m_inputFormat is not supposed to be changed by ADSP. This kind of hard to spot errors happen as a result of poor design. The names of the variable imply their roles inside the object: m_inputFormat -> resampler/remapper -> m_format. with adsp it should look like this:
m_inputFormat -> ADSP -> m_adspFormat -> resampler/remapper -> m_format